### PR TITLE
#12 & #13 Merchant Items Index: 5 most popular items #13. Merchant Items Index: Top Item's Best Day

### LINKED ISSUE #87: 12. Merchant Items Index: 5 most popular items
As a merchant
When I visit my items index page
Then I see the names of the top 5 most popular items ranked by total revenue generated
And I see that each item name links to my merchant item show page for that item
And I see the total revenue generated next to each item name

Notes on Revenue Calculation:
- Only invoices with at least one successful transaction should count towards revenue
- Revenue for an invoice should be calculated as the sum of the revenue of all invoice items
- Revenue for an invoice item should be calculated as the invoice item unit price multiplied by the quantity (do not use the item unit price)

### LINKED ISSUE #88: 13. Merchant Items Index: Top Item's Best Day
As a merchant
When I visit my items index page
Then next to each of the 5 most popular items I see the date with the most sales for each item.
And I see a label “Top selling date for <item name> was <date with most sales>"

Note: use the invoice date. If there are multiple days with equal number of sales, return the most recent day.

### PR DESCRIPTION
**Pull request by z-fitch**
___x_ Wrote Tests __x__ Implemented __x__ Reviewed

12. Merchant Items Index: 5 most popular items

As a merchant
When I visit my items index page
Then I see the names of the top 5 most popular items ranked by total revenue generated
And I see that each item name links to my merchant item show page for that item
And I see the total revenue generated next to each item name

Notes on Revenue Calculation:
- Only invoices with at least one successful transaction should count towards revenue
- Revenue for an invoice should be calculated as the sum of the revenue of all invoice items
- Revenue for an invoice item should be calculated as the invoice item unit price multiplied by the quantity (do not use the item unit price)

13. Merchant Items Index: Top Item's Best Day

As a merchant
When I visit my items index page
Then next to each of the 5 most popular items I see the date with the most sales for each item.
And I see a label “Top selling date for <item name> was <date with most sales>"

Note: use the invoice date. If there are multiple days with equal number of sales, return the most recent day.

Necessary checkmarks:

    [x] All Tests are Passing

    [x] The code will run locally

Type of change

    [x] New feature
    [] Bug Fix

Implements/Fixes:

   Adds the top items for the merchants and shows the top items days
closes #87 
closes #88 

Check the correct boxes

    [] This broke nothing
    [] This broke some stuff
    [] This broke everything

Testing Changes

    [] No Tests have been changed
    [] Some Tests have been changed
    [] All of the Tests have been changed(Please describe what in the world happened)

Checklist:

    [x] My code has no unused/commented out code
    [x] I have reviewed my code
    [x] I have commented my code, particularly in hard-to-understand areas
    [x] I have fully tested my code

(REQUIRED)Please Include a link to a gif of your feelings about this branch

Link: https://media.giphy.com/media/jnQYWZ0T4mkhCmkzcn/giphy.gif

### CODE REVIEW

**Review by Daniel-Gallagher92 — APPROVED** [review]
We worked on this together, and I think we nailed it!

**Review by ethanrossblack — CHANGES_REQUESTED** [review]
just setting to request changes so i can review before anyone merges

File: spec/models/item_spec.rb [hunk 1]
```diff
@@ -35,10 +35,18 @@
...
       end
     end
-  end
 
+    describe ".top_5_items" do 
+      it "returns the top 5 items by total revenue generated" do
+        original_test_data
+        
+        expect(@merchant_1.items.top_5_items).to eq([@item_1, @item_5, @item_7, @item_3, @item_4])
+      end
```
Comment (ethanrossblack) [inline spec/models/item_spec.rb:44]: there are no tests in here that confirm the validity of the total revenue generated. I did some manual calculations for your tests in index_spec, and something is off in your calculations.

**Review by Daniel-Gallagher92 — APPROVED** [review]

File: spec/models/item_spec.rb (hunk 1 above)
Comment (ethanrossblack) [reply spec/models/item_spec.rb:44]: In a perfect world, we should add some tests here for the .top_five_items method that test the total revenue count, but for now, let's just merge it.

**Review by ethanrossblack — APPROVED** [review]

**Pull request merged**